005_team_3_agent_design/005_team_3_agent_design.tex

```diff
@@ -1 +1,129 @@
+
+\subsection{Agent generation}
+%%TODO: have to add the code snippet
+The function generates a new agent by initializing its variables, knowledge and decisions. All knowledge and decisions are defined initially as empty except for our last HP, which is initialized as 100 (as all agents are). \par
+The variables of our agent, which define the personality it has, are randomly allocated to generate different agents. All three variables are defined to be in the range 0 to 100. Therefore morality and mood are randomly allocated to a value between 0 and 100. To ensure our agent is not completely deaf from the start we limit the initial stubbornness to 75. This does not limit the agent once it starts interacting and taking decisions. Which means it could possibly end up with a stubbornness value of 100 if the situations it goes through are auspicious for it.
+
+\subsection{Agent Strategy}
+Our agent follows the following strategy depending on the three agent variables:
+\begin{enumerate}
+    \item When morality is high, agents are more willing to listen and obey other agents.
+    \item When morality is low, agents are less willing to listen and choose to eat more than they would when they have high morality.
+    \item When stubbornness is high, agents will be “deaf” to most messages and not react to them.
+    \item When stubbornness is high, agents are more willing to listen to messages.
+    \item When mood is high, the actions taken are beneficial for more people.
+    \item When mood is low, the actions taken are more beneficial for ourselves.
+\end{enumerate}
+
+\section{Agents emotions}\label{agents_emotions}
+The following functions act as helpers for the rest of the agent code, enabling changes in the agents variables and reactions due to the value of these variables, as well as reactions to friendships. 
+
+\subsection{Read function}
+\texttt{func (a *CustomAgent3) read() bool}: our agent's stubbornness will limit the amount of messages it might read and react to. Each time we receive a message we call the \texttt{read()} function. This function will take into account the current stubbornness of the agent and return if the message must be read, and reacted to, or just discarded. 
+This function works by randomly generating a number between 0 and 100. If this number is higher to the current stubbornness the message is read, if not it will be discarded. 
+
+\subsection{Friendship Function}
+\begin{sloppypar} %%Gives style warning so this sloppypar solves it
+\texttt{func (a *CustomAgent3) updateFriendship(friend uuid.UUID, change int)}: changes the value of the friendships using Q-Learning. Change is either -1 or 1. \par
```

Comment:
similar to earlier comments, uuid.UUID conveys no meaning that UUID doesn't

The reader doesn't need to see the package